src/reviews/views.py

```diff
@@ -114,13 +114,10 @@ def get_context_data(self, **kwargs):
             .values_list('vote')
             .annotate(count=Count('vote'))
         )
-        vote_count_pairs = stage_1_vote_count_pairs.union(stage_2_vote_count_pairs)
-        vote_mapping = collections.defaultdict(
-            (lambda: 0),
-            ((self.vote_keys[k], v)
-             for k, v in vote_count_pairs
-             if k in self.vote_keys),
-        )
+        vote_count_pairs = stage_1_vote_count_pairs.union(stage_2_vote_count_pairs, all=True)
```

Comment:
What does `all=True` change here? Intuitively I’d feel it should be false.

Reply:
@uranusjr , Thank you for your review!

### Reason:
If we don't add `all=True`, we can't include the count of the same level when they are the same in stage 1 and stage  2.

### The description In Django Docs:
The [UNION operator](https://docs.djangoproject.com/en/4.0/ref/models/querysets/#union) selects only distinct values by default. To allow duplicate values, use the `all=True` argument.

---

### For example:
#### 1. Without `all=True`

```bash
# (level, count)
stage 1 stats: ("Strong Accept +1", 1), ("Strong Reject –1", 1) 
stage 2 stats: ("Weak Reject –0", 1), ("Strong Reject –1", 1) 
```
The total stats (`vote_count_pairs`) for this reviewer without `all=True` will be,
``` bash
# <ReviewQuerySet [('-0', 1), ('-1', 1), ('+1', 1)]>
("Weak Reject –0", 1), ("Strong Reject –1", 1), ("Strong Accept +1", 1)
```
which is the result we don't want.


#### 2. With  `all=True`. It can solve the scenario when it happens to have the same value in stage 1 and stage 2.

So if we add `all=True`, it will allow duplicate values and it is what we expect before adding up.

```bash
# <ReviewQuerySet [('-0', 1), ('-1', 1), ('+1', 1), ('-1', 1)]>
("Weak Reject –0", 1), ("Strong Reject –1", 1), ("Strong Accept +1", 1), ("Strong Reject –1", 1)
```


I think this issue can approach with two solutions:

1. This commit:

```python
vote_count_pairs = stage_1_vote_count_pairs.union(stage_2_vote_count_pairs, all=True)
# vote_count_pairs: <ReviewQuerySet [('-0', 1), ('-1', 1), ('+1', 1), ('-1', 1)]>
vote_mapping = collections.defaultdict(int)
for k, v in vote_count_pairs:
    vote_mapping[self.vote_keys[k]] += v
```


2. Alternatively, I found out that it can also approach with `|`:

```python
vote_count_pairs = stage_1_vote_count_pairs | stage_2_vote_count_pairs
# vote_count_pairs: <ReviewQuerySet [('+1', 1), ('-0', 1), ('-1', 2)]>
vote_mapping = collections.defaultdict(
    (lambda: 0),
    ((self.vote_keys[k], v)
     for k, v in vote_count_pairs
     if k in self.vote_keys),
)
```

If there is any misunderstanding, please tell me. Thank you!

Reply:
Ah this is probably due to how Django identifies equality between model objects. OK this makes sense if it is to avoid erroneous row merging.